membership-attribute-service/app/controllers/AttributeController.scala

```diff
@@ -99,4 +120,20 @@ class AttributeController(attributesFromZuora: AttributesFromZuora, commonAction
   def attributes = lookup("attributes", onSuccessMember = identity[Attributes], onSuccessSupporter = identity[Attributes], onNotFound = notFound, sendAttributesIfNotFound = true)
   def features = lookup("features", onSuccessMember = Features.fromAttributes, onSuccessSupporter = _ => Features.unauthenticated, onNotFound = Features.unauthenticated)
 
+
+  def oneOffContributions = {
+    AuthAndBackendViaIdapiAction(Return401IfNotSignedInRecently).async { implicit request =>
+      if (request.redirectAdvice.emailValidated.contains(true)) {
+        authenticationService.userId(request) match {
+          case Some(identityId) =>
+            oneOffContributionDatabaseService.getAllContributions(identityId).map {
+              case -\/(err) => Ok(err)
```

Comment:
was there a reason to return errors as a 200 OK?  Probably shouldn't expose the error messsage in the rresponse though.